extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -66,10 +67,14 @@ void registerVertxGraphqlUI(VertxGraphqlRecorder recorder,
                             + "\", this is not allowed as it blocks the application from serving anything else.");
         }
 
-        Handler<RoutingContext> handler = recorder.handler(path);
-        routes.produce(new RouteBuildItem(path, handler));
-        routes.produce(new RouteBuildItem(path + "/*", handler));
-        displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(path + "/"));
+        Handler<RoutingContext> handler = recorder.handler();
+        routes.produce(nonApplicationRootPathBuildItem.routeBuilder().route(path).handler(handler)
+                .requiresLegacyRedirect()
+                .displayOnNotFoundPage("Graph QL UI", path + "/")
```

Comment:
Not major, but there should not be a space between Graph and QL
```suggestion
                .displayOnNotFoundPage("GraphQL UI", path + "/")
```

Reply:
I can fix that when I do an update at a later PR.

Reply:
We can do it now, as we'll be rolling it back everywhere.